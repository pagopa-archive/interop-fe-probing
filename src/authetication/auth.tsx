import awsmobile from './aws-exports'
import { Auth, Amplify } from 'aws-amplify'

Amplify.configure(awsmobile)

async function login(username: string, password: string) {
  try {
    const user = await Auth.signIn(username, password)
    const token = user.signInUserSession.idToken.jwtToken
    const refreshToken = user.signInUserSession.refreshToken.token
    const accessToken = user.signInUserSession.accessToken.jwtToken
    sessionStorage.setItem('token', token)
    sessionStorage.setItem('refreshToken', refreshToken)
    sessionStorage.setItem('accessToken', accessToken)
    return user
  } catch (error) {
    console.log('error signing in', error)
  }
}

async function logout() {
  try {
    await Auth.signOut()
    sessionStorage.clear()
  } catch (error) {
    console.log('error signing out: ', error)
  }
}

export { login, logout }
