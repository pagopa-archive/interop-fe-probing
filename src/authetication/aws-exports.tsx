/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
  aws_project_region: import.meta.env.VITE_AWS_PROJECT_REGION,
  aws_cognito_identity_pool_id: null,
  aws_cognito_region: import.meta.env.VITE_AWS_COGNITO_REGION,
  aws_user_pools_id: import.meta.env.VITE_AWS_USER_POOL_ID,
  aws_user_pools_web_client_id: import.meta.env.VITE_AWS_USER_POOLS_WEB_CLIENT_ID,
  oauth: {},
  aws_cognito_username_attributes: ['EMAIL'],
  aws_cognito_social_providers: [],
  aws_cognito_signup_attributes: ['EMAIL'],
  aws_cognito_mfa_configuration: 'OFF',
  aws_cognito_mfa_types: [],
  aws_cognito_password_protection_settings: {
    passwordPolicyMinLength: 8,
    passwordPolicyCharacters: [],
  },
  aws_cognito_verification_mechanisms: ['EMAIL'],
}

export default awsmobile
