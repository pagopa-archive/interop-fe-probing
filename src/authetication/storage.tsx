const setStorage = (name: string, value: string) => {
  return new Promise((resolve, reject) => {
    sessionStorage.setItem(name, value);
    return resolve("OK");
  });
};

const deleteStorage = (name: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.removeItem(name));
  });
};

const resetStorage = () => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.clear());
  });
};

const getStorage = (item: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.getItem(item));
  });
};

export { setStorage, deleteStorage, resetStorage, getStorage };
