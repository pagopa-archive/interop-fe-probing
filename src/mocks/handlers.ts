import { rest } from 'msw'

export const handlers = [
  rest.get('http://localhost:3000/eservices', (req, res, ctx) => {
    const mockResponse = {
      content: [
        {
          id: '2',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'OFFLINE',
          versionNumber: 8,
        },
        {
          id: '1',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '3',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '10',
          eserviceName: 'Erogazione borse di studio',
          producerName:
            'A.Di.S.U.R.C. Azienda per Il Diritto allo Studio Universitario della Regione Campania',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '11',
          eserviceName: 'Funzionamento pannelli solari',
          producerName: 'Comune di Brindisi',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '4',
          eserviceName:
            'Posizione centraline ambientali con un nome davvero molto molto molto lungo',
          producerName: 'Comune di Pescara',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '5',
          eserviceName: 'Postazioni e-bike',
          producerName: 'Comune di Roma',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '6',
          eserviceName: 'Stato colonnine di ricarica auto elettriche',
          producerName: 'Comune di Bari',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '7',
          eserviceName: 'Stato lavori Olimpiadi 2026',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '8',
          eserviceName: 'Stato opere PGT 2030',
          producerName: 'Comune di Trento',
          state: 'n/d',
          versionNumber: 1,
        },
      ],
      offset: 0,
      limit: 10,
      totalElements: 11,
    }

    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/producers', (req, res, ctx) => {
    const mockResponse = [
      {
        label: 'producer2',
        value: 'producer2',
      },
    ]
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/mainData/', (req, res, ctx) => {
    const mockResponse = {
      eserviceName: 'Probing test 3',
      producerName: 'Comune di Milano',
      versionNumber: '7',
      pollingFrequency: 5,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/probingData/', (req, res, ctx) => {
    const mockResponse = {
      probingEnabled: true,
      state: 'offline',
      responseReceived: '2023-04-27T14:55:25Z',
      eserviceActive: false,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/telemetryData/eservices/', (req, res, ctx) => {
    const mockResponse = {
      values: [
        {
          responseTime: 640,
          status: 'OK',
          time: '2023-05-16T13:17:01.579Z',
        },
        {
          responseTime: 98,
          status: 'OK',
          time: '2023-05-16T13:33:07.271Z',
        },
        {
          responseTime: 6,
          status: 'OK',
          time: '2023-05-16T13:39:01.371Z',
        },
      ],
      percentages: [
        {
          value: 94.28572,
          status: 'OK',
        },
        {
          value: 5.714286,
          status: 'KO',
        },
      ],
    }
    return res(ctx.json(mockResponse))
  }),
]
