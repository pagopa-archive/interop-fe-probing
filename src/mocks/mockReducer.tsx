import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

function reducer(ui: any, { preloadedState, store, ...renderOptions }: any = {}) {
  function Wrapper({ children }: any) {
    return (
      <Router>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
      </Router>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export { reducer }
