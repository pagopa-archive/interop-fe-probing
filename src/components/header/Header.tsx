import { HeaderAccount, HeaderProduct, ProductSwitchItem, RootLinkType } from '@pagopa/mui-italia'
import { useNavigate } from 'react-router-dom'
import { logout } from '../../authetication/auth'
import { useTranslation } from 'react-i18next'
import stores from '../../store/Store'

const link: RootLinkType = {
  title: 'PagoPA S.p.A.',
  label: 'PagoPA S.p.A.',
  href: 'https://www.pagopa.it/it/',
  ariaLabel: '',
}

const productsList: ProductSwitchItem[] = [
  {
    id: 'pdnd',
    title: 'PDND InteroperabilitÃ ',
    productUrl: '',
    linkType: 'internal',
  },
]

const Header = () => {
  const { t } = useTranslation(['general', 'loginPage'])

  const navigate = useNavigate()

  const [updateSnackbar] = stores.useSnackbarStore((state) => [state.updateSnackbar])

  const [updateLogStatus] = stores.useLogStatusStore((state) => [state.updateLogStatus])

  const handleLogout = async () => {
    logout()
      .then((res) => {
        updateLogStatus(false)
        navigate('/login')
        updateSnackbar(true, t('logoutSuccessMessage', { ns: 'loginPage' }), 'success')
      })
      .catch((error) => {
        updateSnackbar(true, t('errorRequest', { ns: 'general' }), 'error')
      })
  }

  return (
    <header>
      <HeaderAccount
        enableAssistanceButton={false}
        loggedUser={stores.useLogStatusStore((state) => state.status)}
        rootLink={link}
        onAssistanceClick={() => {
          console.log('Clicked/Tapped on Assistance')
        }}
        onLogin={() => navigate('/login')}
        onLogout={() => handleLogout()}
      />
      <HeaderProduct productsList={productsList} />
    </header>
  )
}
export default Header
