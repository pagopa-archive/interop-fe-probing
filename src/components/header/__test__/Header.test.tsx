import { describe, expect, test, vi } from 'vitest'
import userEvent from '@testing-library/user-event'
import { screen, cleanup, act, renderHook } from '@testing-library/react'
import Header from '../Header'
import { renderWithRouterAndQuery } from '../../../mocks/mockWrapper'
import * as router from 'react-router'
import * as auth from '../../../authetication/auth'

describe('Header', () => {
  afterEach(cleanup)

  test('render component', () => {
    const { container } = renderWithRouterAndQuery(<Header />)
    expect(container).toMatchSnapshot()
  })

  test('click Accedi button', async () => {
    const navigate = vi.fn()
    vi.spyOn(router, 'useNavigate').mockImplementation(() => navigate)

    renderWithRouterAndQuery(<Header />)
    const button = screen.getByRole('button', { name: 'Accedi' })
    const user = userEvent.setup()
    await user.click(button)
    expect(navigate).toBeCalled()
  })

  test('click Esci button and simulate success', async () => {
    vi.spyOn(Storage.prototype, 'getItem').mockImplementation(() => {
      return 'asd'
    })

    vi.spyOn(auth, 'logout').mockImplementation(async () => {
      return Promise.resolve()
    })

    const navigate = vi.fn()
    vi.spyOn(router, 'useNavigate').mockImplementation(() => navigate)

    renderWithRouterAndQuery(<Header />)

    const button = screen.getByRole('button', { name: 'Esci' })
    const user = userEvent.setup()
    await user.click(button)
    expect(navigate).toBeCalled()
  })

  test('click Esci button and simulate error', async () => {
    vi.spyOn(auth, 'logout').mockImplementation(async () => {
      throw new Error('UNKNOWN ERROR')
    })

    vi.spyOn(Storage.prototype, 'getItem').mockImplementation(() => {
      return 'asd'
    })

    const navigate = vi.fn()
    vi.spyOn(router, 'useNavigate').mockImplementation(() => navigate)

    renderWithRouterAndQuery(<Header />)

    const button = screen.getByRole('button', { name: 'Esci' })
    const user = userEvent.setup()
    try {
      await user.click(button)
    } catch (error: unknown) {
      if (error instanceof Error) {
        expect(error.message).toBe('UNKNOWN ERROR')
      }
    }
  })
})
