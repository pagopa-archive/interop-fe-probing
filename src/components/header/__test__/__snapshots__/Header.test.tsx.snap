// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > render component 1`] = `
<div>
  <header>
    <div
      class="MuiStack-root css-1fmjzam-MuiStack-root"
    >
      <div
        class="MuiContainer-root css-ur2jdm-MuiContainer-root"
      >
        <div
          class="MuiStack-root css-hxrfoi-MuiStack-root"
        >
          <a
            aria-label=""
            class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall css-1074q1t-MuiButtonBase-root-MuiButton-root"
            href="https://www.pagopa.it/it/"
            rel="noreferrer"
            tabindex="0"
            target="_blank"
            title="PagoPA S.p.A."
          >
            PagoPA S.p.A.
          </a>
          <div
            class="MuiStack-root css-yh6xev-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              title="Accedi"
              type="button"
            >
              Accedi
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1u2a3dz"
    >
      <div
        class="MuiContainer-root css-ur2jdm-MuiContainer-root"
      >
        <div
          class="MuiStack-root css-176y9rv-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1yo1jo6-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-sx5zcp-MuiTypography-root"
            >
              PDND Interoperabilità
            </p>
          </div>
          <div
            class="MuiBox-root css-jta2rl"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header > render component 1`] = `
<div>
  <header>
    <div
      class="MuiStack-root css-1fmjzam-MuiStack-root"
    >
      <div
        class="MuiContainer-root css-ur2jdm-MuiContainer-root"
      >
        <div
          class="MuiStack-root css-hxrfoi-MuiStack-root"
        >
          <a
            aria-label=""
            class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-root MuiButton-naked MuiButton-nakedText MuiButton-sizeSmall MuiButton-nakedSizeSmall css-1074q1t-MuiButtonBase-root-MuiButton-root"
            href="https://www.pagopa.it/it/"
            rel="noreferrer"
            tabindex="0"
            target="_blank"
            title="PagoPA S.p.A."
          >
            PagoPA S.p.A.
          </a>
          <div
            class="MuiStack-root css-yh6xev-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              title="Accedi"
              type="button"
            >
              Accedi
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1u2a3dz"
    >
      <div
        class="MuiContainer-root css-ur2jdm-MuiContainer-root"
      >
        <div
          class="MuiStack-root css-176y9rv-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1yo1jo6-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-sx5zcp-MuiTypography-root"
            >
              PDND Interoperabilità
            </p>
          </div>
          <div
            class="MuiBox-root css-jta2rl"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
